recipes/disopred/addcmdarguments.patch

```diff
@@ -0,0 +1,178 @@
+ ##
+ ## Please refer to the license agreement for permission of use and re-distribution
+ ##
++## Stefan Janssen: patched this script to make it more convenient to the user, by
++## adding arguments to: 
++##	  1) specify blast database: -d
++##	  2) output directory: -o 
++##	  3) blast binary directory: -n
++##	  4) resume computation from pre-computed PSIBLAST search: -mtx
+ 
+ use strict;
+ use File::Basename;
+ use Cwd qw(abs_path);
+-
++use Getopt::Long;
++use Data::Dumper;
+ 
+ ## IMPORTANT: Set the paths to folder with the NCBI executables and to the
+ ## sequence database
+@@ -32,41 +39,54 @@
+ 
+ my $DISO2_FPR = 5; # the adjustable DISOPRED2 false positive rate, represented as an integer between 1 and 10
+ 
++my $mtx = 0;
++my $outdir = undef;
++&get_arguments();
++
+ # check whether the command line contains only one argument
+-die "[$0] ERROR: The command line should only contain the path to the input sequence file\n" if ( scalar @ARGV != 1);
++die "[$0] ERROR: The command line should only contain the path to the input sequence file or mtx file\n" if ( scalar @ARGV != 1);
+ 
+ # die if input file does not exist or is not a text file
+ die "[$0] ERROR: Input file $ARGV[0] does not exist\n"  if !-e $ARGV[0];
+ die "[$0] ERROR: Input file $ARGV[0] does not look like a text file\n"  if !-T $ARGV[0];
+ 
+ my $fasta_fn = abs_path($ARGV[0]);
++my $mtx_fn = $fasta_fn;
+ my ($out_dir, $base) = (dirname($fasta_fn), basename($fasta_fn));
++$out_dir = $outdir if (defined $outdir);
+ 
+ $base =~ s/\.fa(sta)?$//;
++$base =~ s/\.mtx$//;
++
+ my $host_id = `hostid`;
+ chomp $host_id;
+ my $tmp_base = join '_', $base , $$, $host_id;
+ 
+-my ($hits_file, $chk_file) = map {my $name = join '.', $tmp_base, $_; join '/', $out_dir, $name } ("blast", "chk");
+-
+-print "Running PSI-BLAST search ...\n\n";
+-# run psiblast
+-my $args = join ' ', $NCBI_DIR."blastpgp", "-i", $fasta_fn, "-d", $SEQ_DB, $PSIBLAST_PAR, "-C", $chk_file, "-o", $hits_file, "\n";
+-system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
+-
+-print "Generating PSSM ...\n\n";
+-
+-$args = join ' ', "echo", $chk_file, ">", $tmp_base.".pn", "\n";
+-system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
+-
+-$args = join ' ', "echo", $fasta_fn, ">", $tmp_base.".sn", "\n";
+-system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
+-
+-$args = join ' ', $NCBI_DIR."makemat", "-P", $tmp_base, "\n";
+-system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
+-
+-my $mtx_fn = join '/', $out_dir, $tmp_base.".mtx";
+-die "[$0] ERROR: Couldn't find the mtx file $mtx_fn\n" if !-e $mtx_fn;
++my ($args, $hits_file, $chk_file) = (undef, "", "");
++if ($mtx == 0) {
++	($hits_file, $chk_file) = map {my $name = join '.', $tmp_base, $_; join '/', $out_dir, $name } ("blast", "chk");
++
++	print "Running PSI-BLAST search ...\n\n";
++	# run psiblast
++	$args = join ' ', $NCBI_DIR."blastpgp", "-i", $fasta_fn, "-d", $SEQ_DB, $PSIBLAST_PAR, "-C", $chk_file, "-o", $hits_file, "\n";
```

Comment:
This works but is bit confusing. `join(" ", ($NCBI_DIR."blastpgp", "-i", $fasta_fn, ...))` may sound cleaner. If that is the style of the original codebase, then never mind.